text_generation/causal_lm/cpp/chat_model_lm.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <iostream>
+
+#include "group_beam_searcher.hpp"
+#include "openvino/openvino.hpp"
+
+namespace {
+std::pair<ov::Tensor, ov::Tensor> tokenize(ov::InferRequest& tokenizer, std::string&& prompt) {
+    constexpr size_t BATCH_SIZE = 1;
+    tokenizer.set_input_tensor(ov::Tensor{ov::element::string, {BATCH_SIZE}, &prompt});
+    tokenizer.infer();
+    return {tokenizer.get_tensor("input_ids"), tokenizer.get_tensor("attention_mask")};
+}
+
+std::string detokenize(ov::InferRequest& detokenizer, const std::vector<int64_t>& tokens) {
+    constexpr size_t BATCH_SIZE = 1;
+    ov::Tensor inp = detokenizer.get_input_tensor();
+    inp.set_shape({BATCH_SIZE, tokens.size()});
+    for (size_t idx = 0; idx < tokens.size(); ++idx) {
+        inp.data<int64_t>()[idx] = tokens.at(idx);
+    }
+    detokenizer.infer();
+    return detokenizer.get_output_tensor().data<std::string>()[0];
+}
+}  // namespace
+
+std::string generate_chat_promt_gemma(const std::string& input) {
+    std::stringstream result_promt;
+    result_promt << "<bos><start_of_turn>user\n" << input << "<end_of_turn>\n<start_of_turn>model";
+    return result_promt.str();
+}
+
+int main(int argc, char* argv[]) try {
+    if (argc != 2) {
+        throw std::runtime_error(std::string{"Usage: "} + argv[0] + " <MODEL_DIR> ");
+    }
+    // Compile models
+    ov::Core core;
+    core.add_extension(OPENVINO_TOKENIZERS_PATH);  // OPENVINO_TOKENIZERS_PATH is defined in CMakeLists.txt
+    // tokenizer and detokenizer work on CPU only
+    ov::InferRequest tokenizer =
+        core.compile_model(std::string{argv[1]} + "/openvino_tokenizer.xml", "CPU").create_infer_request();
+    ov::InferRequest detokenizer =
+        core.compile_model(std::string{argv[1]} + "/openvino_detokenizer.xml", "CPU").create_infer_request();
+    // The model can be compiled for GPU as well
+    ov::InferRequest lm =
+        core.compile_model(std::string{argv[1]} + "/openvino_model.xml", "CPU").create_infer_request();
+
+    int64_t total_positions = 0;
+    int32_t global_beam_idx = 0;
+
+    std::string promt;
+    for (;;) {
+        std::cout << "User promt:\n";
+        std::getline(std::cin, promt);
+        std::cout << "\n";
+
+        if (!promt.compare("Stop!"))
```

Comment:
Looks like we need to print somewhere at the beginning, that `Stop!` is a keyword.

Reply:
done